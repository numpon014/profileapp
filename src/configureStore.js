import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'connected-react-router';
import thunk from 'redux-thunk';
import createReducer from './reducers';

export default function configureStore(initialState, history) {
  const middlewares = [thunk, routerMiddleware(history)];
  const middlewareEnhancer  = [applyMiddleware(...middlewares)];

  return createStore(
    createReducer(),
    initialState,
    compose(...middlewareEnhancer),
  );
}
