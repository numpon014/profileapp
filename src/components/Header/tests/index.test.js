import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { createMemoryHistory } from 'history';

import Header from '../index';
import configureStore from '../../../configureStore';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({t: key => key})
}));

describe('<Header />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should render className parameter', () => {
    const className = "main-header"
    const { container } = render(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <Header className={className} />
        </ConnectedRouter>
      </Provider>,
    );
    expect(container.querySelector(`div.${className}`)).not.toBeNull();
  });

  it('should render logo and home link in nav', () => {
    const { container } = render(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <Header />
        </ConnectedRouter>
      </Provider>,
    );

    expect(container.querySelector(`a.home-link`)).not.toBeNull();
    expect(container.querySelector(`img.logo`)).not.toBeNull();
  });
});
