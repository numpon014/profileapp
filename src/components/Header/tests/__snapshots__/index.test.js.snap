// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render a div 1`] = `
<div
  class="sc-bdvvtL cBjQmZ"
>
  <nav
    class="navbar navbar-expand navbar-light bg-light"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        <img
          alt="app.logo"
          class="logo"
          src="logo.png"
        />
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          class="me-auto navbar-nav"
        >
          <a
            class="home-link nav-link"
            href="/"
          >
            app.header.navbar.home
          </a>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
